### DISCUSSION TITLE
Timefold pricing

### DISCUSSION THREAD

**Discussion by mcimbora** [TimefoldAI/timefold-solver#55] [General]
Hi team,

We're experimenting with OptaPlanner -> Timefold migration with one of the customers (employee rostering UC). So far we've seen impressive score calculation speedups.

We use multithreaded solving and judging by this pull request https://github.com/TimefoldAI/timefold-solver/pull/53 we won't be able to do so any longer with the Community edition. Do you have plans to publish a transparent pricing model for the Enterprise edition? What are the variables included in estimating the customer costs?

Thanks,
Matej

**Comment by ge0ffrey** [comment]
We're a young company and we're currently still figuring out fair pricing for the paid edition. Bear with us.
We understand this is a financial risk for projects upgrading to Timefold that require one of the 2 high-scalability features (multithreaded solving or nearby selection).
Contact us at info@timefold.ai to mitigate this risk.

The 0.9.x releases (currently 09.38 and 0.9.39) still have those 2 high-scalability features (for practical reasons), as will all future 0.9.x releases, but 1.0.0 won't as mentioned in [the fork message](https://timefold.ai/blog/2023/optaplanner-fork/). Pull request 53 doesn't take them away yet, but does prepare that work towards 1.0.0.

We're currently still researching the details on how to best deliver that paid jar with the least amount of friction (for example through a no-login public maven repo with "source available", free for development used, paid for production use).

**Comment by mcimbora** [comment]
Thank you @ge0ffrey, it makes sense. We'll get in touch 👍

**Comment by holgerbrandl** [comment]
We're also experimenting with OptaPlanner -> Timefold migration. Clearly, figuring out a solid business model for a dual-license framework is a hard decision. However, taking multithreaded optimization out of the core,  essentially leaves users of the open-source version with an API for single-threaded exploration which is not really useable in production (for which solvers at least must scale out on a single node imho as all other solver-libs do). I personally find this a quite harsh measure, which may significantly affect community adoption.

As stated above, I also think that the current lack of a transparent pricing scheme does not really yet welcome serious users.

**Comment by ge0ffrey** [comment]
Hi Holger,

Timefold Community still supports solving multiple datasets (SolverManager), each on their own thread, on a single node. You'll need Timefold Enterprise to solve a single dataset with multiple threads (moveThreadCount).
In my experience, few others solvers support this and only 10% of OptaPlanner users use this high-scalability feature. More importantly, these users are typically enterprises that benefit highly from OptaPlanner, exactly the ones we 'd like to have chip in for the development costs.

I wish we could have kept Timefold entirely open source, but after investing all business models, it was clear that Open Core (with only a few features not being Open Source) was the only way to guarantee the continuation of the OptaPlanner project.

**Comment by holgerbrandl** [comment]
Thanks for your kind reply.
